source/blog/2017-12-08-host-deploy-customization.html.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Customizing the host deploy process
+author: omachace
+tags: Ansible, Automation, Configuration management, host-deploy, firewalld
+date: 2017-12-08 15:00:00 CET
+comments: true
+published: true
+---
+
+In 4.2 release we have introduced a possibility to customize the host-deploy proccess by running the Ansible post-tasks after the host-deploy process successfully finishes.
+
+READMORE
+
+## The reason
+Prior to 4.2 release when user need to customize the IPTables configuration of the hosts, he had to do database hacking. In 4.2 we have introduced the support for FirewallD,
+and we wanted to add users the ability to customize the FirewallD configuration as well. Most people didn't like the database modification, so we have introduced Ansible integration,
+which is much more easier and powerfull than the database modification.
+
+## The host-deploy playbook
+Currently when the host is added to the engine or reinstalled, we execute the host-deploy playbook
+located at `/usr/share/ovirt-engine/playbooks/ovirt-host-deploy.yml`.
+
+This playbook executes few internal roles to configure for example OVN driver, libvirt guests or FirewallD.
+
+## Special tasks file
+
+As part of this role we also include additional tasks, which could be written by user, to modify the host-deploy
+process for example to open some more FirewallD ports.
+
+Those additional tasks can be added to following file:
+
+```
+/etc/ovirt-engine/ansible/ovirt-host-deploy-post-tasks.yml
+```
+
+## Example
+An example post-tasks file is provided by `ovirt-engine-tools` RPM, at following location:
```

Comment:
Why is it provided by ovirt-engine-tools? Tools don't have anything in common with Ansible support, no?

Reply:
We provide there all the playbooks and *sh scripts:

 https://github.com/oVirt/ovirt-engine/blob/master/ovirt-engine.spec.in#L1121

Reply:
OK, I somehow missed that during reviews, `ovirt-engine-tools` package should provide only command line tools. But I'd rather not mention the package name directly (if we move playbooks into different package), I'd just mention that this playbook is provided as a part of ovirt-engine installation

Reply:
done